plugin-procfs/README.md

```diff
@@ -1,3 +1,112 @@
 # Procfs plugin
 
-Collects processes- and system-related metrics by reading the `/proc` virtual filesystem.
+Collects processes and system-related metrics by reading the [proc](<https://www.kernel.org/doc/html/latest/filesystems/proc.html>) virtual filesystem on UNIX based operating systems.
+
+## Requirements
+
+- UNIX operating system.
+- All metrics of the plugin needs to access to the process file system `/proc/sys/` dynamically generated by the kernel.
```

Comment:
What is `/proc/sys`? We aren't reading from that.

What the plugin needs is read access to `/proc`. Depending on the situation, it will monitor more or less processes. For instance, running the Alumet agent as root will allow the procfs plugin to monitor every process of every user. Running the Alumet agent as user `gerard` will make it monitor the processes of this user. The permission to read `/proc/pid` for other processes can also be granted with capabilities. See https://docs.kernel.org/filesystems/proc.html#process-specific-subdirectories

For this PR, it would be great to make a separate section about permissions and capabilities, and put only one sentence in the list item.

Reply:
To be more exhaustive, `/proc` access primarily depends on the `hidepid` mount option: https://docs.kernel.org/filesystems/proc.html#mount-options